src/Tables.jl

```diff
@@ -256,24 +636,573 @@ function Dict2LatexTable(d::Dict, refs::Dict)
+        "ϕ"=>"Friction angle *(°)*",
+        "ψ"=>"Dilation angle *(°)*",
+        "C"=>"Cohesion *(Pa)*",
+        "Vp"=>"P-wave velocity *(km/s)*",
+        "Vs"=>"S-wave velocity *(km/s)*",
+        "T0"=>"Reference temperature *(°C)*",
+        "P0"=>"Reference pressure *(Pa)*",
+        "β"=>"Compressibility *(1/Pa)*",
+        "α"=>"Thermal expansion coeff. *(1/K)*"
+        )
+
+    # Generates latex preamble
+    Table = " | Denotation | Variablename | "
+
+    # Creates table headers
+    counter = 1
+    for i = 1:parse(Int64, d["Name 1"][2])
+        Table *=  d["Name $i"][1] * " | "
+    end
+
+    Table *= "\n"
+    Table *= " | ---------- | ------------ | "
+
+    for i = 1:parse(Int64, d["Name 1"][2])
+        Table *= "-" ^ length(d["Name $i"][1]) * " | "
+    end
+
+    Table *= "\n"
+
+    # Get vector with all unique symbols without phasenames
+    for key in 1:length(dictpairs)
+        dictpairs_key = dictpairs[key].first
+        if occursin("Name", dictpairs_key)
+            continue
+        # Checks if symbol is from field CompositeRheology and if the symbol occurs more often than once
+        elseif (occursin("R", dictpairs_key[1:3]))
+            push!(symbs, dictpairs[key].second[2])
+        elseif maximum(occursin.(["CompositeRheology", "CreepLaws"], dictpairs_key)) && !(occursin("R", dictpairs_key[1:3]))
+            if occursin("Disl", dictpairs[key].second[6])
+                number = parse(Int64, dictpairs[key].second[5])
+                push!(symbs, "A_disl" * "^$number")
+            elseif occursin("Diff", dictpairs[key].second[6])
+                number = parse(Int64, dictpairs[key].second[5])
+                push!(symbs, "A_diff" * "^$number")
+            else
+                number = parse(Int64, dictpairs[key].second[5])
+                push!(symbs, dictpairs[key].second[2] * "$number")
+            end
+        else
+            push!(symbs, dictpairs[key].second[2])
+        end
+    end
+    symbs = unique(sort(symbs))
+
+    # Creates columnwise output for all parameters of the input phase
+    for symbol in symbs
+        # Sets parametername and variable
+        if length(symbol) > 1
+            # Checks if Unicode chars in combination with a number etc. (e.g "α0") are used which apparently do not have a second index but only first and third
+            # and checks if symbol has another number in it than 0
+            if maximum(endswith.(symbol,["0","1","2","3","4","5","6","7","8","9"])) && !occursin("_", symbol)
+                if length(unidecode(symbol)) > 2 && occursin("0", symbol)
+                    Table *= " " * string(desc[symbol]) * " | " * symbol[1] * "~" * symbol[3] * "~"
+                elseif length(unidecode(symbol)) <= 2 && occursin("0", symbol)
+                    Table *= " " * string(desc[symbol]) * " | " * symbol[1] * "~" * symbol[2] * "~"
+                elseif length(unidecode(symbol)) > 2 && maximum(occursin.(["1","2","3","4","5","6","7","8","9"], symbol))
```

Comment:
`maximum(occursin(string(i), symbol) for i in 1:9)` is more idiomatic and does not allocate an intermediate array